common/utils.cpp

```diff
@@ -74,8 +74,8 @@ bool guessResolution(const char* filename, int& w, int& h)
     int state = STATE_START;
     std::string strFilename(filename);
     const char* tokStart = NULL;
-    long width = 0;
-    long hight = 0;
+    long long width = 0;
+    long long hight = 0;
```

Comment:
Line 97 and line 119 should use `std::strtoll`

Reply:
...and line 132 should use `long long maxInt`

Reply:
@xuguangxin `long` was added originally in a088df6e6f762a27f7b13b8342260ad100920fbf to detect overflow of w,h output vars on 64 bit sys.

Reply:
...that is, if user provided filename has WxH > INT_MAX, then it is invalid.

Reply:
but for video encoder or vaCreateSurface, we can't create such surface. 16K may enough for many years

Reply:
Right.  But this function is only responsible for parsing the width and height from user specified filename string (no knowledge of va capabilities).  We can't assume user always specifies good input filename.  The `long long width` and `long long hight` are only temporary local variables to read the width and height from the filename string for validating.  Consider if we used std::atoi when parsing the width and height from filename string... then if width and height are > INT_MAX, the result would be undefined.  I suppose we could still have the same issue if width and height in filename is > maximum long long.  Nonetheless, I never really liked how this function was implemented and would be nice to refactor to a better solution.

Reply:
@uartie , if under layer returns an undefined value for this, no matter how to handle it in up layer. it still has the problem.
instead of use a088df6 maybe b3c3086582b695bd036628fc0f7dc260de10ba3d in #838 ,  it only change few lines and it can catch some exceptions. 
the target of this guessResolution function is provide some handle for us. 
When you do the vpp or decode, you get the yuv, but it's no place to get the original resolution. you will forget it in someday.  With this function, we can write the resolution down in file name. 
The technical in this function called [DFA](http://goldparser.org/articles/dfa.htm), most of the compiler use it to collect tokens